### PR TITLE
Allows drones to switch to/from harm intent

### LINKED ISSUE #25292: Drones can't weld airlocks
Recent changes require harm intent to weld airlocks, this makes repairing an emagged airlock not possible by simple mobs.

### PR DESCRIPTION
**Pull request by RandomMarine**
:cl: RandomMarine
add: Drones can now switch between help and harm intent.
/:cl:
How convenient that the drone UI already supports the intent button.

For standard law-abiding drones, this fixes #25292 and similar unreported issues.
For rogue drones, harm intent helps them harm better.

### CODE REVIEW

**Review by Cyberboss — APPROVED** [review]

**Comment by Cyberboss** [comment]
Drones having harm intent seems very backwards tbh though

**Comment by RandomMarine** [comment]
I'm not letting them punch/kick anything, though.

Ultimately, I'd like if drones had their own UI icons that had an image of a drone fixing something for the HELP icon and a drone welding an airlock for the HARM icon, to make their purposes more clear. But I lack the artistic ability for this.

**Comment by Cyberboss** [comment]
Actually, does harm intent always damage the airlock? I know I started a campaign of "If you're on harm intent you will aggressively hit the thing with the thing"

**Comment by RandomMarine** [comment]
If you're using a welder, harm intent makes you weld it closed. (As opposed to just repairing it)

But there's some other useful applications for harm intent. One example being that it's impossible to use a labeler on a locker without harm.

**Comment by MimicFaux** [comment]
Yaaaasss

**Comment by Anonmare** [comment]
You also need to be able to weld an airlock shut to deconstruct it without an RCD.

Plus it'd be a good idea to give them intent now in case other things start requiring different intents for the purpose of repair/construction

File: code/modules/mob/living/silicon/silicon_defense.dm [hunk 1]
```diff
@@ -77,6 +77,11 @@
 				"<span class='warning'>[M] punches [src], but doesn't leave a dent.</span>", null, COMBAT_MESSAGE_RANGE)
 	return 0
 
+/mob/living/silicon/attack_drone(mob/living/simple_animal/drone/M)
+	if(M.a_intent == INTENT_HARM)
+		return
+	..()
```
Comment (lzimann) [inline code/modules/mob/living/silicon/silicon_defense.dm]: `return ..()`

File: code/modules/mob/living/simple_animal/animal_defense.dm [hunk 2]
```diff
@@ -80,6 +80,11 @@
 			damage = rand(20, 35)
 		return attack_threshold_check(damage)
 
+/mob/living/simple_animal/attack_drone(mob/living/simple_animal/drone/M)
+	if(M.a_intent == INTENT_HARM) //No kicking dogs even as a rogue drone. Use a weapon.
+		return
+	..()
```
Comment (lzimann) [inline code/modules/mob/living/simple_animal/animal_defense.dm]: `return ..()`

**Review by lzimann — CHANGES_REQUESTED** [review]

**Review by lzimann — APPROVED** [review]

**Pull request merged**